### ISSUE TITLE
[LIB REQUEST]: 

### ISSUE THREAD

**Issue by FCLC** [compiler-explorer/compiler-explorer#3426]
### Library name

OpenBLAS

### Library description

OpenBLAS is an optimized BLAS (Basic Linear Algebra Subprograms) library based on GotoBLAS2 1.13 BSD version.

### Library repository

https://github.com/xianyi/OpenBLAS

### Motivation

I've been attempting to teach the usage of basic HPC libraries, of which OpenBLAS is one of the largest. Serving as one of the easiest to use references for linear algebra implementations in SIMD, it also serves to teach how seemingly serial code can be vectorized.

**Comment by jeremy-rifkin** [comment]
I'd like to see OpenBLAS supported in CE too. If someone does not beat me to it in the next few days I'll give a stab at a PR.

**Comment by FCLC** [comment]
> I'd like to see OpenBLAS supported in CE too. If someone does not beat me to it in the next few days I'll give a stab at a PR.

Yeah, getting a BLAS library in would be great, perhaps PETSC down the line could also be a nice addition.

**Comment by jeremy-rifkin** [comment]
OpenBLAS is a challenging library to incorporate on a tool like CE because it is very architecture-specific.

I went through the process of figuring out how to add OpenBLAS to the config and ce-infra but I'm realizing even with `-DDYNAMIC_ARCH=TRUE` an architecture-specific `config.h` is required. Currently https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/internal/AddingLinkableLibrary.md reads:

> 5. If a config header file has been generated based on the compiler configuration, we can only maybe support this if the header does NOT include any defines about the architecture.

So I think this might be a bit of a dead-end. @partouf if you have time to weigh in it would be appreciated!

I'm suspecting that this might be a situation where it could theoretically be supported but would require a lot of work to do so.
